ui/dialog.js

```diff
@@ -872,4 +869,29 @@ return $.widget( "ui.dialog", {
 	}
 });
 
+// DEPRECATED
+if ( $.uiBackCompat !== false ) {
+
+	// DialogClass option
+	$.widget( "ui.dialog", $.ui.dialog, {
+		options: {
+			dialogClass: ""
+		},
+		_createWrapper: function() {
+			this._super();
+			this.uiDialog.addClass( this.options.dialogClass );
+		},
+		_setOption: function( key, value ) {
+			if ( key === "dialogClass" ) {
+				this.uiDialog
+					.removeClass( this.options.dialogClass )
+					.addClass( value );
+			}
+			this._superApply( arguments );
+		}
+	});
+}
+
+return $.ui.dialog;
```

Comment:
Let's add a TODO to switch this back when removing back compat because this is easily overlooked. Could just be added as a note on the DEPRECATED comment to make it obvious.